tests/concurrent_test.go

```diff
@@ -71,3 +72,73 @@ func TestProxy_Concurrency(t *testing.T) {
 	}
 	wg.Wait()
 }
+
+func TestAgent_MultipleConn(t *testing.T) {
```

Comment:
Can you add a comment about the purpose of this test, something like
"This test verifies that when one stream between a proxy agent and the proxy server terminates, the proxy server does not terminate frontends supported by the same proxy agent but different streams."